proposals/jib_core_library.md

```diff
@@ -0,0 +1,74 @@
+# Proposal: Jib Core as a library for Java
+
+The tracking issue is at [#337](https://github.com/GoogleContainerTools/jib/issues/337).
+
+# Goal
+
+Design for Jib Core as a Java library for building container images.
+
+# Proposed API
+
+`Jib` - the main entrypoint for using Jib Core
+- `JibContainerBuilder from(String imageReference)`
+- `JibContainerBuilder from(ImageReference imageReference)`
+
+`JibContainerBuilder` - configures the container to build
+- `JibContainerBuilder addLayer(List<Path> files, Path pathInContainer)`
+- `JibContainerBuilder setLayers(List<LayerConfiguration>/LayerConfiguration...)`
+- `JibContainerBuilder addLayer(LayerConfiguration)`
+- `JibContainerBuilder setEntrypoint(List<String>/String...)`
+- `JibContainerBuilder setProgramArguments(List<String>/String...)`
+- `JibContainerBuilder setEnvironment(Map<String, String> environmentMap)`
+- `JibContainerBuilder addEnvironmentVariable(String name, String value)`
+- `JibContainerBuilder setExposedPorts(List<Port>/Port...)`
+- `JibContainerBuilder addExposedPort(Port port)`
+- `JibContainerBuilder setLabels(Map<String, String> labelMap)`
+- `JibContainerBuilder addLabel(String key, String value)`
```

Comment:
Weren't we talking about having a more fluent-looking `environment()`, `labels()`?

Reply:
Right, that was in #953 - I'll update these to reflect the proposal there.

Reply:
But I don't think we ever replace layers, do we.  At least, not for the API. Right?

Reply:
`setLayers` currently would actually replace layers (https://github.com/GoogleContainerTools/jib/blob/master/jib-core/src/main/java/com/google/cloud/tools/jib/api/JibContainerBuilder.java#L109)